src/Aspire.Dashboard/Components/Controls/ResourceDetails.razor.cs

```diff
@@ -0,0 +1,158 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using Aspire.Dashboard.Model;
+using Aspire.Dashboard.Otlp.Model;
+using Microsoft.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components;
+
+namespace Aspire.Dashboard.Components.Controls;
+
+public partial class ResourceDetails
+{
+    [Parameter, EditorRequired]
+    public required ResourceViewModel Resource { get; set; }
+
+    [Parameter, EditorRequired]
+    public Dictionary<OtlpApplication, int>? UnviewedErrorCounts { get; set; }
+
+    [Parameter]
+    public bool ShowSpecOnlyToggle { get; set; }
+
+    private bool IsSpecOnlyToggleDisabled => Resource.Environment.Any(i => i.FromSpec == false) is false;
+
+    private bool _showAll;
+
+    private IQueryable<EnvironmentVariableViewModel> FilteredItems =>
+        Resource.Environment.Where(vm =>
+            (_showAll || vm.FromSpec) &&
+            (vm.Name.Contains(_filter, StringComparison.CurrentCultureIgnoreCase) ||
+            vm.Value?.Contains(_filter, StringComparison.CurrentCultureIgnoreCase) == true)
+        ).AsQueryable();
+
+    private IQueryable<Endpoint> FilteredEndpoints => GetEndpoints().AsQueryable();
+
+    private IQueryable<SummaryValue> FilteredSummaryValues => GetSummaryValues().AsQueryable();
+
+    private string _filter = "";
+    private bool _defaultMasked = true;
+
+    private readonly Icon _maskIcon = new Icons.Regular.Size16.EyeOff();
+    private readonly Icon _unmaskIcon = new Icons.Regular.Size16.Eye();
+    private readonly Icon _showSpecOnlyIcon = new Icons.Regular.Size16.DocumentHeader();
+    private readonly Icon _showAllIcon = new Icons.Regular.Size16.DocumentOnePage();
+
+    private readonly GridSort<EnvironmentVariableViewModel> _nameSort = GridSort<EnvironmentVariableViewModel>.ByAscending(vm => vm.Name);
+    private readonly GridSort<EnvironmentVariableViewModel> _valueSort = GridSort<EnvironmentVariableViewModel>.ByAscending(vm => vm.Value);
+
+    private IEnumerable<Endpoint> GetEndpoints()
+    {
+        foreach (var endpoint in Resource.Endpoints)
+        {
+            if (endpoint.EndpointUrl.Contains(_filter, StringComparison.CurrentCultureIgnoreCase))
+            {
+                yield return new Endpoint { Name = "Endpoint Url", IsHttp = true, Address = endpoint.EndpointUrl };
```

Comment:
If the name is for display, it should be localised.

Reply:
Will do localization once UI is agreed on.